src/irs/core/interrogator.py

```diff
@@ -241,37 +223,57 @@ async def interrogate(  # noqa: PLR0913
         upload_date: str,
         decrypted_size: int,
         sha256_checksum: str,
+        s3_endpoint_alias: str,
     ):
         """
         Forwards first file part to encryption key store, retrieves file encryption
         secret(s) (K_data), decrypts file and computes checksums. The object and bucket
         ID parameters refer to the object_id and bucket_id associated with the upload,
         i.e. not the staging bucket.
         """
-        object_size = await get_object_size(
+        try:
+            staging_bucket_id, object_storage = self._object_storages.for_alias(
+                s3_endpoint_alias
+            )
+        except KeyError as error:
+            raise ValueError(
+                f"S3 endpoint alias not configured: {s3_endpoint_alias}"
+            ) from error
+
+        object_size = await object_storage.get_object_size(
             object_id=source_object_id, bucket_id=source_bucket_id
         )
         part_size = calc_part_size(file_size=object_size)
+
+        # generate ID for the staging bucket file
+        object_id = str(uuid.uuid4())
+        staging_ids = S3IDs(bucket_id=staging_bucket_id, object_id=object_id)
+        inbox_ids = S3IDs(bucket_id=source_bucket_id, object_id=source_object_id)
+
+        object_storage_handler = StagingHandler(
```

Comment:
You have defined the StagingHandler as a 3hex translator (part of the adapter), however, that translator should always be constructed as part of dependency injection. The core should retrieve a ready to use instance of the translator that complies with the Port definition. In general, the core should not be dependent on the adapter implementation only on the ports.